authors/apps/articles/views.py

```diff
@@ -48,17 +68,64 @@ def update(self, request, slug, *args, **kwargs):
             raise exceptions.PermissionDenied()
             
         new_article = request.data.get("article", {})
+
         serializer = self.serializer_class(
             article_to_update, data=new_article, partial=True
         )
+
+        serializer_context = {'request': request}
+
+        serializer = self.serializer_class(
+            article_to_update, 
+            data=new_article, 
+            partial=True,
+            context=serializer_context
+        )
         serializer.is_valid(raise_exception=True)
         serializer.save()
         return Response(serializer.data, status=status.HTTP_200_OK)
 
     def retrieve(self, request, slug):
+        serializer_context = {'request': request}
         try:
             article = Article.objects.get(slug=slug)
         except Article.DoesNotExist:
             raise Http404
+        serializer = self.serializer_class(article, context=serializer_context)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class ArticleFavouriteAPIView(APIView):
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ArticleJSONRenderer,)
+    serializer_class = ArticleSerializer
+
+    def post(self, request, slug):
+        profile = request.user.profile
+        serializer_context = {'request': request}
+
+        try:
+            article = Article.objects.get(slug=slug)
+        except Article.DoesNotExist:
+            raise Http404
+
+        profile.favourite(article)
+
+        serializer = self.serializer_class(article, context=serializer_context)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+    def delete(self, request, slug):
+        profile = request.user.profile
+        serializer_context = {'request': request}
+
+        try:
```

Comment:
This `try....except...` block that you use to check for the existence of an article appears in both the `post` and `delete` methods. Wouldn't it be better to move this bit of code into its own method in this class, so as to make our `post` and `delete` methods look **cleaner** by removing the duplication?

Reply:
The duplication has been fixed.